test/test_issues.jl

```diff
@@ -0,0 +1,15 @@
+include("common.jl")
+
+MPI.Init(; threadlevel=:single)
+
+@testset "#688" begin
```

Comment:
This is redundant if you restore `Char` in `MPIDatatype` (which I this point I believe is the right thing to do), the same code is executed in `test_allgather.jl` already.

Reply:
I prefer to be explicit about it, but your call. Removed.